business/dashboards.go

```diff
@@ -383,6 +383,10 @@ func (in *DashboardsService) buildLabelsQueryString(namespace string, labelsFilt
 	for k, v := range labelsFilters {
 		labels += fmt.Sprintf(`,%s="%s"`, prometheus.SanitizeLabelName(k), v)
 	}
+	for labelName, labelValue := range in.promConfig.QueryScope {
+		labels += fmt.Sprintf(`,%s="%s"`, labelName, labelValue)
```

Comment:
What is this `prometheus.SanitizeLabelName(k)` I see above in line 384 - and do you need to sanitize the `labelName` here as well? i.e. should this be:
```suggestion
		labels += fmt.Sprintf(`,%s="%s"`, prometheus.SanitizeLabelName(labelName), labelValue)
```

Reply:
I saw that as well but I don't think it's applicable/needed.  The labels used here must already be valid, as they are custom and being added in the target environment to the scraped metrics.  That sanitization was needed because of the scenario in https://github.com/kiali/kiali/issues/3737.

Reply:
Anytime I see the word "sanitized" in the context of software, I immediately think it is trying to prevent a security loophole (like [sanitizing input to avoid a log4j CVE](https://www.whitesourcesoftware.com/resources/blog/best-practices-for-dealing-with-log4j/) :) ) 

Are we sure this isn't needed to sanitize the labels for something security related? See the code introduced in the scenario / PR you refer to: https://github.com/kiali/kiali/pull/3739/files#diff-a21497378081b5b2b2cf631b8cf327bd5f7905b48e3fdb34518e1503ef7a6713R309-R310
```
// SanitizeLabelName replaces anything that doesn't match invalidLabelCharRE with an underscore.
// Copied from https://github.com/prometheus/prometheus/blob/df80dc4d3970121f2f76cba79050983ffb3cdbb0/util/strutil/strconv.go
func SanitizeLabelName(name string) string {
	return invalidLabelCharRE.ReplaceAllString(name, "_")
}
```

Reply:
Fine, I'll add it, although the chance of malicious behavior here seems very limited.